test/decrypter.test.js

```diff
@@ -1,5 +1,153 @@
 'use strict';
-var loadManifest = require('../src/js/decrypter.js'),
-    Q = QUnit;
 
-Q.module('Decrypter');
+var Q = QUnit,
+    videojs = window.videojs,
+    Decrypter = require('../src/js/decrypter.js'),
+    player,
+    video,
+    decrypter,
+    keyRequests,
+    keys,
+    trigger;
+
+trigger = function(el, event) {
+  var k, domEvent = new Event(event.type);
+  for (k in event) {
+    if (k === 'type') {
+      continue;
+    }
+    domEvent[k] = event[k];
+  }
+  el.dispatchEvent(domEvent);
+};
+
+Q.module('Decrypter', {
+  setup: function() {
+    video = document.createElement('video');
+    video.onneedkey = null;
+    video.onwebkitneedkey = null;
+    video.canPlayType = function() {
+      return true;
+    };
+    // mock out the EME 0.1b key event flow
+    keyRequests = [];
+    video.webkitGenerateKeyRequest = function(keySystem, initData) {
+      keyRequests.push(Array.prototype.slice.call(arguments));
+    };
+    keys = [];
+    video.webkitAddKey = function(keySystem, key, initData, sessionId) {
+      keys.push(Array.prototype.slice.call(arguments));
+    };
+    player = videojs(video);
+    decrypter = new Decrypter(player.tech);
+  }
+});
+
+Q.test('errors if an unsupported key system is detected', function() {
+  var errors = [];
+  player.on('error', function(event) {
+    errors.push(event);
+  });
+  video.canPlayType = function() {
+    return false;
+  };
+  video.webkitGenerateKeyRequest = function() {
+    errors.push('should not request keys when an error occurs');
+  };
+  trigger(video, {
+    type: 'webkitneedkey',
+    keySystem: 'unsupported key system'
+  });
+
+  Q.equal(errors.length, 1, 'triggered an error');
+  Q.equal(player.error().code,
+          window.MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED,
+          'type is SRC_NOT_SUPPORTED');
+});
+
+Q.test('throws early if used in a browser without EME support', function() {
+  try {
```

Comment:
QUnit has a [`throws`](http://api.qunitjs.com/throws/) method that can be used for this.